io_scene_godot/converters/simple_nodes.py

```diff
@@ -20,31 +21,88 @@ def export_empty_node(escn_file, export_settings, node, parent_gd_node):
     return empty_node
 
 
+class CameraNode(NodeTemplate):
+    """Camera node in godot scene"""
+    attribute_conversion = [
+        # blender attr, godot attr, converter lambda, type
+        ('clip_end', 'far', lambda x: x, float),
+        ('clip_start', 'near', lambda x: x, float),
+        ('angle', 'fov', math.degrees, float),
+        ('ortho_scale', 'size', lambda x: x, float),
+    ]
+
+    def __init__(self, name, parent):
+        super().__init__(name, "Camera", parent)
+
+
 def export_camera_node(escn_file, export_settings, node, parent_gd_node):
     """Exports a camera"""
     if (node.data is None or node.hide_render or
             "CAMERA" not in export_settings['object_types']):
         return parent_gd_node
 
-    cam_node = NodeTemplate(node.name, "Camera", parent_gd_node)
+    cam_node = CameraNode(node.name, parent_gd_node)
     camera = node.data
 
-    cam_node['far'] = camera.clip_end
-    cam_node['near'] = camera.clip_start
+    for item in cam_node.attribute_conversion:
+        blender_attr, gd_attr, converter, _ = item
+        cam_node[gd_attr] = converter(getattr(camera, blender_attr))
 
     if camera.type == "PERSP":
         cam_node['projection'] = 0
-        cam_node['fov'] = math.degrees(camera.angle)
     else:
         cam_node['projection'] = 1
-        cam_node['size'] = camera.ortho_scale
 
     cam_node['transform'] = fix_directional_transform(node.matrix_local)
     escn_file.add_node(cam_node)
 
+    export_animation_data(escn_file, export_settings,
+                          cam_node, node.data, 'camera')
+
     return cam_node
 
 
+class LightNode(NodeTemplate):
+    """Base class for godot light node"""
+    attribute_conversion = [
+        ('use_specular', 'light_specular', lambda x: 1.0 if x else 0.0, bool),
+        ('energy', 'light_energy', lambda x: x, float),
+        ('color', 'light_color', mathutils.Color, mathutils.Color),
+        ('shadow_color', 'shadow_color', mathutils.Color, mathutils.Color),
+    ]
+
+
+class OmniLightNode(LightNode):
+    """OminiLight node in godot"""
+    attribute_conversion = LightNode.attribute_conversion + [
+        ('distance', 'omni_range', lambda x: x, float)
+    ]
+
+    def __init__(self, name, parent):
+        super().__init__(name, "OmniLight", parent)
+
+
+class SpotLightNode(LightNode):
```

Comment:
There are now lots of small classes. I can understand why you did this, but perhaps the light node should now be moved into a separate file? (Or perhaps there is another way that doesn't use so many classes)

Reply:
I will try to do it in a single class,  I would like still put light nodes in the `simple_nodes.py`.